### ISSUE TITLE
Performance and battery consumption

### ISSUE THREAD

**Issue by defagos** [SRGSSR/playsrg-ios#91]
Having a look at recent reviews:

> **Akku Killer (Hintergrundaktivität)**  ·  Grundsätzlich solides App. Man bezahlt ja auch Billag, da darf man sowas erwarten. Jedoch saugt die Hintergrundaktivität massiv Akku... wieso eigentlich? Ich musste es deshalb jedenfalls löschen.

> **Energiefresser**  ·  Benötigt etwa 15% Akku in 30 Minuten. Handy wird warm. Bei älteren Versionen hatte ich dieses Problem nicht. Bitte App reparieren!

Users are complaining about performance issues and battery life.

No significant changes have been made recently which could seriously have impacted battery life (we namely did not add any potentially CPU-intensive feature in the last months).

Historically, I must admit we encountered heating battery issues with some devices, especially old ones (with diminished battery health). I also encountered issues of a similar nature with media playback in Safari on such devices. We know that the app should be optimized (see #9, #22, #70, #127, #132), but sadly didn't have enough time to analyze and tackle all issues.

Part of the reason for these battery issues is probably in media playback as well. This is CPU-consuming and cannot be entirely eliminated, especially for video. We can at least have a look at other background activity (data refreshes, image refreshes, etc.) to see whether CPU pressure is somehow too high. I would recommend profiling the current Letterbox demo application, then remove all periodic updates and profile it again. This way we would have a better understanding if there is something we can really do. Profiling SRG Media Player would be also a good indicator.

Note that with other devices (iPhone X / XS, for example) I did not encounter such issues, but of course these devices are beasts. My feeling is that the device battery health is probably a factor, and that maybe recent changes in iOS 12 have changed how battery is consumed while playing.

**Comment by defagos** [comment]
Having a look at [`AVRouteDetector`](https://developer.apple.com/documentation/avfoundation/avroutedetector/2915762-routedetectionenabled?language=objc) documentation, which replaces `MPVolumeView` since iOS 11 (and which itself is deprecatd as of iOS 13), I am asking myself whether `MPVolumeView` itself can drain battery.

As we are replacing `MPVolumeView` for iOS 11 and above (hopefully using `AVRouteDetector` in a correct way), let's have a look at battery consumption again.

**Comment by defagos** [comment]
New tools available in Xcode 11 organizer should provide us with interesting insights about battery consumption. The corresponding talk is available [here](ttps://developer.apple.com/wwdc19/417). The examples are probably a bit too simple to be honest, still it can be interesting to have an overall idea of the changes between versions.

**Comment by defagos** [comment]
I further reduced the energy consumption by [reducing the need for timers in Letterbox](https://github.com/SRGSSR/srgletterbox-apple/issues/199). 

Moreover, I ensured timers which must never run in background (e.g. those related to UI refreshes) are suspended. I created a `SmartTimer` class for this purpose, which could later be part of a general library (but this is an entirely different topic).

 In general, when we need some periodic process to occur, we can decide between three kinds of timers based on what is needed:

- Timers bound to playback: Use a periodic time observer. Example: Time slider refreshes.
- Timers which are not directly bound to playback, but which should continue to run if the app is active because it is playing some content: Use `NSTimer`. Example: Periodic synchronization of user data.
- Timers which are not directly bound to playback and which should not continue to run in background, even if the app is playing soe content. Use `SRGTimer`. Example: Periodic UI refreshes on live cells.
